cvmfs/publish/cmd_enter.cc

```diff
@@ -574,6 +580,40 @@ int CmdEnter::Main(const Options &options) {
     if (rvi != 0)
       throw EPublish("cannot chroot to " + rootfs_dir_);
     LogCvmfs(kLogCvmfs, kLogStdout, "done");
+
+    SettingsBuilder builder;
+    UniquePtr<Publisher> publisher;
+
+    if (options.Has("transaction")) {
+      LogCvmfs(kLogCvmfs, kLogStdout,
+               "Starting a transaction inside the enter shell");
+
+      if (options.Has("repo-config")) {
+        repo_config_ = options.GetString("repo-config");
+        LogCvmfs(kLogCvmfs, kLogStdout,
+                 "Parsing the external configuration for the repository at %s",
+                 repo_config_.c_str());
+
+        std::string config;
+        std::string config_file = repo_config_ + "/" + fqrn_ + "/server.conf";
+        std::string folderpath = session_dir_ + "/" + fqrn_;
+        MkdirDeep(folderpath.c_str(), 0600, true /* veryfy_writable */);
+
+        std::string session_config_file = folderpath + "/server.conf";
+        int fd_config = open(config_file.c_str(), O_RDONLY);
+        SafeReadToString(fd_config, &config);
+        SafeWriteToFile(config, session_config_file, 0600);
+
+        builder.SetConfigPath(session_dir_);
+      }
+
+      SettingsPublisher *settings_publisher =
+          builder.CreateSettingsPublisher(fqrn_, false);
+      publisher = new Publisher(*settings_publisher);
+      publisher->session()->SetKeepAlive(true);
+      publisher->Transaction();
```

Comment:
Couldn't `Transaction()` call `session()->SetKeepAlive(true)` internally?

Reply:
Yep, I think that's right!  We can call `SetKeepAlive(true)` at the end of the `Transaction()` method (in the success code path). That would also fix a dangling issue here: if the transaction fails, the session won't get cleaned up.